src/server/vfs/mac/litesyncextconnector.mm

```diff
@@ -1362,19 +1364,52 @@ bool checkIoErrorAndLogIfNeeded(IoError ioError, const std::string &itemType, co
 
         if (isSyncing != isSyncingCurrent || isHydrated != isHydratedCurrent) {
             // Update parent directory status
-            QFileInfo info(path);
-            return vfsProcessDirStatus(info.dir().path());
+            const QString parentPath = QFileInfo(path).dir().path();
+            if (parentPath == _localSyncPath) return true;
+
+            if (isSyncing) {
+                {
+                    const std::lock_guard<std::mutex> lock(_mutex);
+                    _syncingFolders[parentPath].insert(path);
+                }
+                vfsSetStatus(parentPath, isSyncing, 100, isHydrated);
+            }
+            else {
+                _mutex.lock();
+                _syncingFolders[parentPath].remove(path);
+                if (_syncingFolders[parentPath].size() == 0) {
+                    _syncingFolders.remove(parentPath);
+                    _mutex.unlock();
+
+                    if (!vfsProcessDirStatus(parentPath)) {
```

Comment:
Should we adapt the other calls to `vfsProcessDirStatus` in this file?

Reply:
Looks like the other call does not have any performance issue. I would prefer keep the changes to a minimum on this branch if the test does not show any problem with this implementation. What do you think?